src/block/utils/CompostFactory.php

```diff
@@ -0,0 +1,187 @@
+		$this->register(VanillaBLOCKS::BROWN_MUSHROOM_BLOCK()->asItem(), 65);
+		$this->register(VanillaBLOCKS::RED_MUSHROOM_BLOCK()->asItem(), 65);
+
+		$this->register(VanillaBlocks::NETHER_WART()->asItem(), 65);
+		$this->register(VanillaItems::POTATO(), 65);
+		$this->register(VanillaBlocks::PUMPKIN()->asItem(), 65);
+		$this->register(VanillaBlocks::CARVED_PUMPKIN()->asItem(), 65);
+		$this->register(VanillaBlocks::SEA_PICKLE()->asItem(), 65);
+		$this->register(VanillaBlocks::SHROOMLIGHT()->asItem(), 65);
+		$this->register(VanillaBlocks::SPORE_BLOSSOM()->asItem(), 65);
+		$this->register(VanillaItems::WHEAT(), 65);
+		//Crimson fungus
+		//Warped fungus
+		//Crimson roots
+		//Warped roots
+
+		// region: 85% percentage compost
+		$this->register(VanillaItems::BAKED_POTATO(), 85);
+		$this->register(VanillaItems::BREAD(), 85);
+		$this->register(VanillaItems::COOKIE(), 85);
+		//Flowering Azalea
+		$this->register(VanillaBlocks::HAY_BALE()->asItem(), 85);
+		$this->register(VanillaBlocks::NETHER_WART_BLOCK()->asItem(), 85);
+		//Pitcher Plant
+		//Torchflower
+		$this->register(VanillaBlocks::WARPED_WART_BLOCK()->asItem(), 85);
+
+		// region: 100% percentage compost
+		$this->register(VanillaBlocks::CAKE()->asItem(), 100);
+		$this->register(VanillaItems::PUMPKIN_PIE(), 100);
+	}
+
+	private function registerFlowers() : void{
+		$this->register(VanillaBlocks::DANDELION()->asItem(), 50);
+
+		$this->register(VanillaBlocks::POPPY()->asItem(), 50);
+		$this->register(VanillaBlocks::BLUE_ORCHID()->asItem(), 50);
+		$this->register(VanillaBlocks::ALLIUM()->asItem(), 50);
+		$this->register(VanillaBlocks::AZURE_BLUET()->asItem(), 50);
+		$this->register(VanillaBlocks::RED_TULIP()->asItem(), 50);
+		$this->register(VanillaBlocks::ORANGE_TULIP()->asItem(), 50);
+		$this->register(VanillaBlocks::WHITE_TULIP()->asItem(), 50);
+		$this->register(VanillaBlocks::PINK_TULIP()->asItem(), 50);
+		$this->register(VanillaBlocks::OXEYE_DAISY()->asItem(), 50);
+		$this->register(VanillaBlocks::CORNFLOWER()->asItem(), 50);
+		$this->register(VanillaBlocks::LILY_OF_THE_VALLEY()->asItem(), 50);
+
+		$this->register(VanillaBlocks::SUNFLOWER()->asItem(), 65);
+		$this->register(VanillaBlocks::LILAC()->asItem(), 65);
+		$this->register(VanillaBlocks::ROSE_BUSH()->asItem(), 65);
+		$this->register(VanillaBlocks::PEONY()->asItem(), 65);
+	}
+
+	public function register(Item $item, int $percentage, bool $overwrite = false) : bool{
+		if($percentage < 0 || $percentage > 100){
+			throw new \InvalidArgumentException("Percentage must be in range " . self::MIN_PERCENTAGE . " ... " . self::MAX_PERCENTAGE);
+		}
+		$typeId = $item->getTypeId();
+		if(($overwrite || !isset($this->list[$typeId])) && !$item->isNull()){
+			$this->list[$typeId] = $percentage;
+			return true;
+		}
+		return false;
+	}
+
+	public function isCompostable(Item $item) : bool{
+		return !$item->isNull() && isset($this->list[$item->getTypeId()]);
+	}
+
+	public function getPercentage(Item $item) : int{
```

Comment:
Might be better to return null instead of 0 when the percentage doesn't exists.

Reply:
Three choices: 
- We return null because the item is not present, so I wouldn't be shocked if the value didn't exist?
- Return 0, but then it would have to be documented?
- Keep it like that because the logic is ok?